mep/footnotes/models.py

```diff
@@ -285,6 +288,36 @@ def index_data(self):
         return index_data
 
 
+class FootnoteQuerySet(models.QuerySet):
+    '''Custom :class:`models.QuerySet` for :class:`Footnote`'''
+
+    def event_date_range(self):
+        '''Find earliest and latest dates for any events associated
+        with footnotes in this queryset. Returns a tuple of earliest
+        and latest dates, or None if no dates are found.'''
+
+        # use get model to avoid circular import
+        Event = apps.get_model('accounts', 'Event')
+        event_content_types = ContentType.objects.filter(
+            app_label='accounts',
+            model__in=['Event', 'Borrow', 'Purchase'])
+
+        # get a list of event ids from the current footnote queryset
+        event_ids = self.filter(content_type__in=event_content_types) \
+                        .values_list('object_id', flat=True)
+        # find corresponding events, filter out unknown years,
+        # and aggregrate dates to get earliest and latest from this set
+        values = Event.objects.filter(pk__in=event_ids).known_years() \
```

Comment:
`values` seems a little generic, but I guess it's pretty obvious what kind of values they are. `date_values` or `dates`?